modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/expression/FilterByCountIndividualSegmentsExpressionVisitorImpl.java

```diff
@@ -214,29 +255,43 @@ public JSONObject toJSONObject() {
 
 		public static class Day {
 
-			public Day(String operator, String value) {
+			public Day(String operator, List<String> value) {
 				_operator = operator;
-				_value = value;
+
+				_values = value;
 			}
 
 			public String getOperator() {
 				return _operator;
 			}
 
-			public String getValue() {
-				return _value;
+			public List<String> getValues() {
+				return _values;
 			}
 
 			public JSONObject toJSONObject() {
+				if (_values.size() == 1) {
+					return JSONUtil.put(
+						"operatorName", _operator
+					).put(
+						"value", _values.get(0)
+					);
+				}
+
 				return JSONUtil.put(
 					"operatorName", _operator
 				).put(
-					"value", _value
+					"value",
+					JSONUtil.put(
+						"end", _values.get(0)
```

Comment:
@claraizquierdo

What do you think that instead receive a end and start value you receive an array?

Reply:
Hello!

I think that the object with `start` and `end` keys provides more semantic, but I can also work with an array being the first element the start date and the second the en date.